AIP-39: ``DagRun.data_interval_start`` and ``data_interval_end`` (#16352)
mistercrunch/Airflow: docs/apache-airflow/macros-ref.rst

=====================================   ====================================
Variable                                Description
=====================================   ====================================
``{{ logical_date }}``                  the logical date of the DAG run (`pendulum.Pendulum`_)

> ``{{ ds }}``                            the logical date as ``YYYY-MM-DD``

Reviewer: I think this should be derrived from data_interval_start, rather than logical date.

Previously the implicit data interval was exection_date...next_exectuion_date

Author: How about `@once` DAGs? We decided to set its data interval to None, should this also become None?

Reviewer: Hmmmmmm, _Probably_, but would make it breaking change :(

Reviewer: Maybe we need to leave this as it is but deprecate it, and replace with `{{ data_interval_start | ds }}` (i.e. add new `ds`, `ds_nodash`, `ts`, etc filters) -- separate PR there.

``{{ ds_nodash }}``                     the logical date as ``YYYYMMDD``

> ``{{ ts }}``                            same as ``schedule_date.isoformat()``. Example: ``2018-01-01T00:00:00+00:00``

Reviewer: ```suggestion
``{{ ts }}``                            same as ``logical_date.isoformat()``. Example: ``2018-01-01T00:00:00+00:00``
```

But see previous comment about deriving form data interval.

``{{ ts_nodash }}``                     same as ``ts`` without ``-``, ``:`` and TimeZone info. Example: ``20180101T000000``
``{{ ts_nodash_with_tz }}``             same as ``ts`` without ``-`` and ``:``. Example: ``20180101T000000+0000``
``{{ next_logical_date }}``             the next logical date if exists (`pendulum.Pendulum`_ or ``None``)
``{{ next_ds }}``                       the next logical date as ``YYYY-MM-DD``
                                        if ``{{ ds }}`` is ``2018-01-01`` and ``schedule_interval`` is ``@weekly``,
                                        ``{{ next_ds }}`` will be ``2018-01-08``

> ``{{ next_ds_nodash }}``                the next schedule date as ``YYYYMMDD`` if exists, else ``None``

Reviewer: ```suggestion
``{{ next_ds_nodash }}``                the next logical date as ``YYYYMMDD`` if exists, else ``None``
```

``{{ execution_date }}``                the execution date (logical date), same as ``logical_date``
``{{ prev_ds }}``                       the previous execution date as ``YYYY-MM-DD``
                                        if ``{{ ds }}`` is ``2018-01-08`` and ``schedule_interval`` is ``@weekly``,
                                        ``{{ prev_ds }}`` will be ``2018-01-01``
``{{ prev_ds_nodash }}``                the previous execution date as ``YYYYMMDD`` if exists, else ``None``
``{{ yesterday_ds }}``                  the day before the execution date as ``YYYY-MM-DD``
``{{ yesterday_ds_nodash }}``           the day before the execution date as ``YYYYMMDD``
``{{ tomorrow_ds }}``                   the day after the execution date as ``YYYY-MM-DD``
``{{ tomorrow_ds_nodash }}``            the day after the execution date as ``YYYYMMDD``
``{{ prev_execution_date }}``           the previous execution date (if available) (`pendulum.Pendulum`_)
``{{ prev_execution_date_success }}``   execution date from prior successful dag run,
                                        same as ``prev_logical_date_success``

> ``{{ next_execution_date }}``           the next execution date, same as ``next_logical_date``

Reviewer: ```suggestion
``{{ next_execution_date }}``           the next execution date, same as ``data_interval_end``
```

(Yes, it would be the same, but I think we should point people to use the explicit data interval properties)
